File: 2026/aed-1/cormen/2.1/2.1-5/addBinaryIntegers.java
public class addBinaryIntegers {
    public static int[] solve(int[] A, int[] B, int n) {
        int[] C = new int[n];
        int sum = 0;

        for (int i = 0; i < n; i++) {
            sum += A[i] * i << i;
            sum += B[i] * i << i;
        }

        for (int i = 0; sum != 0; i++) {
            C[i] = sum % 2;
            sum = sum / 2;
        }

        return C;
    }
    public static void main(String[] args) {
        int[] A = {1, 0, 0, 0, 0};
        int[] B = {1, 1, 0, 0, 0};
        int n = 5;
        int[] C = solve(A, B, n);

        for (int i = 0; i < n; i++) {
            IO.println(C[i]);
        }
    }
}

File: 2026/aed-1/cormen/2.1/2.1-5/add_binary_integers.rs
fn add_binary_integers(a: &[i32], b: &[i32], n: usize) -> Vec<i32> {
    let mut c = vec![0; n + 1];
    let mut sum = 0;

    for i in 0..n - 1 {
        sum += a[i] * 1 << i;
        sum += b[i] * 1 << i;
    }

    let mut i = 0;

    while sum != 0 {
        c[i] = sum % 2;
        sum /= 2;
        i += 1;
    }

    return c;
}

fn main() {
    let a = [1, 0, 0, 0, 0];
    let b = [1, 1, 0, 0, 0];
    let n = 5;
    let c = add_binary_integers(&a, &b, n);

    for i in 0..n - 1 {
        println!("{}", c[i]);
    }
}
